### PR TITLE
[Feature][ConnectorV2] add [File]excel sink

### PR DESCRIPTION
**Pull request by Bingz2**
## Purpose of this pull request
add [File]excel sink
https://github.com/apache/incubator-seatunnel/issues/1946

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml [hunk 1]
```diff
@@ -93,5 +93,18 @@
...
+            <version>4.1.2</version>
+            <scope>compile</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.apache.poi</groupId>
+            <artifactId>poi-ooxml</artifactId>
+            <version>4.1.2</version>
+        </dependency>
+
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml]: Versions should be managed centrally in root-pom.

**Comment by CalvinKirs** [comment]
Check style fail. you can run `./mvnw checkstyle:check` check in your local env.

File: seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml (hunk 1 above)
Comment (Bingz2) [reply seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml]: Excuse me, do you know what causes this?

image

Comment (CalvinKirs) [reply seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml]: 😄 Don't worry.The CodeQL is unstable

File: seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml [hunk 2]
```diff
@@ -93,5 +93,16 @@
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.poi</groupId>
+            <artifactId>poi</artifactId>
+            <scope>compile</scope>
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-file/connector-file-base/pom.xml]: You can delete this, the scope should be managed by the `dependency manager` in the root directory, and the default `compiler`

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/FileFormat.java [hunk 3]
```diff
@@ -55,7 +55,10 @@ public ReadStrategy getReadStrategy() {
         public ReadStrategy getReadStrategy() {
             return new JsonReadStrategy();
         }
-    };
+    },
+    EXCEL("xlsx")
+    ;
+
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/FileFormat.java]: hi, you should add a `ExcelReadStrategy`

File: seatunnel-connectors-v2/connector-file/connector-file-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hdfs/sink/writer/HdfsTransactionStateFileWriteFactory.java [hunk 4]
```diff
@@ -98,8 +99,8 @@ public static TransactionStateFileWriter of(@NonNull SeaTunnelRowType seaTunnelR
                     rowDelimiter,
                     fileSystem);
         }
-        // if file type not supported by file connector, default txt writer will be generated
-        return new HdfsTxtTransactionStateFileWriter(
+        if (fileFormat.equals(FileFormat.EXCEL)) {
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-file/connector-file-hadoop/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/hdfs/sink/writer/HdfsTransactionStateFileWriteFactory.java]: In fact, you don't need to do this, we can automatically choose the format based on the type

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/FileFormat.java (hunk 3 above)
Comment (Bingz2) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/FileFormat.java]: Isn't `ExcelReadStrategy` added by `Excel Source`?

**Comment by hailin0** [comment]
Can you add connector-v2 e2e-testcase?

reference：
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2
https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-spark-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2

**Comment by Bingz2** [comment]
> Can you add connector-v2 e2e-testcase?
> 
> reference： https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2 https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-spark-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2

Okay, I'll add it later

**Review by EricJoy2048 — DISMISSED** [review]
Please add `excel` support to `docs/en/connector-v2/sink/LocalFile.md` and `docs/en/connector-v2/sink/HdfsFile.md`.

**Comment by Bingz2** [comment]
> 

I've already added it

**Comment by Bingz2** [comment]
> Can you add connector-v2 e2e-testcase?
> 
> reference： https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2 https://github.com/apache/incubator-seatunnel/tree/dev/seatunnel-e2e/seatunnel-spark-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/spark/v2

I've already added it

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/resources/file/fakesource_to_hdfs_excel.conf [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+sink {
+  HdfsFile {
+    path="/tmp/hive/warehouse/test2"
+    partition_by=["age"]
+    partition_dir_expression="${k0}=${v0}"
+    is_partition_field_write_in_file=true
+    file_name_expression="${transactionId}_${now}"
+    file_format="json"
```
Comment (TyrantLucifer) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/resources/file/fakesource_to_hdfs_excel.conf]: Do you mean `excel`?

**Review by TyrantLucifer — CHANGES_REQUESTED** [review]

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/resources/file/fakesource_to_hdfs_excel.conf (hunk 5 above)
Comment (Bingz2) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/resources/file/fakesource_to_hdfs_excel.conf]: Yes.... let me change it

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/FileFormat.java [hunk 6]
```diff
@@ -55,7 +55,10 @@ public ReadStrategy getReadStrategy() {
         public ReadStrategy getReadStrategy() {
             return new JsonReadStrategy();
         }
-    };
+    },
+    EXCEL("xlsx")
+    ;
```
Comment (CalvinKirs) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/FileFormat.java]:
``` EXCEL("xlsx") {
        @Override
        public ReadStrategy getReadStrategy() {
            return new ExcelReadStrategy();
        }
```
hi, suggestion change to it, and you alson need excelReadStragey

File: pom.xml [hunk 7]
```diff
@@ -983,6 +985,20 @@
                 <artifactId>commons-net</artifactId>
                 <version>${commons-net.version}</version>
             </dependency>
+
+            <dependency>
+                <groupId>org.apache.poi</groupId>
+                <artifactId>poi</artifactId>
+                <version>${poi.version}</version>
+                <scope>compile</scope>
```
Comment (CalvinKirs) [inline pom.xml]: we can delete this line. the default scope is compile

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/FileFormat.java (hunk 6 above)
Comment (Bingz2) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/FileFormat.java]: image
Do I need to implement it？I see people in the community working on Excel Source

Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/FileFormat.java]: > image Do I need to implement it？I see people in the community working on Excel Source

I don't think you need to implement it in this pr.

**Pull request closed without merging**